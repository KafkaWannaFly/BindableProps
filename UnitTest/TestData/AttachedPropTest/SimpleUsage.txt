
namespace MyMauiApp.Controls
{
    public partial class MyMauiControl : ContentView
    {
        [AttachedProp]
        private bool _hasShadow;
    
        public MyMauiControl()
        {
            // My logic
        }
    }
    
}