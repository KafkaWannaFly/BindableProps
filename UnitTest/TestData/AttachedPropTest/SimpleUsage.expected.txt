// <auto-generated/>

namespace MyMauiApp.Controls
{
    public partial class MyMauiControl
    {
        public static readonly BindableProperty HasShadowProperty = BindableProperty.CreateAttached(
            "HasShadow",
            typeof(bool),
            typeof(MyMauiControl),
            default,
            (BindingMode)0,
            null,
            (bindable, oldValue, newValue) => 
                        ((MyMauiControl)bindable).HasShadow = (bool)newValue,
            null,
            null,
            null
        );
        
        public static bool GetHasShadow(BindableObject view)
        {
            return (bool)view.GetValue(HasShadowProperty);
        }

        public static void SetHasShadow(BindableObject view, bool value)
        {
            view.SetValue(HasShadowProperty, value);
        }
    }
}